service/src/main/scala/za/co/absa/loginsvc/rest/AuthManagerConfig.scala

```diff
@@ -18,36 +18,44 @@ package za.co.absa.loginsvc.rest
 
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.context.annotation.{Bean, Configuration}
-import org.springframework.security.authentication.AuthenticationManager
+import org.springframework.security.authentication.{AuthenticationManager, AuthenticationProvider}
 import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder
 import org.springframework.security.config.annotation.web.builders.HttpSecurity
-import za.co.absa.loginsvc.rest.config.{ActiveDirectoryLDAPConfig, UsersConfig}
+import za.co.absa.loginsvc.rest.config.auth.{ActiveDirectoryLDAPConfig, DynamicAuthOrder, UsersConfig}
 import za.co.absa.loginsvc.rest.provider.ConfigUsersAuthenticationProvider
 import za.co.absa.loginsvc.rest.provider.ad.ldap.ActiveDirectoryLDAPAuthenticationProvider
 
+import scala.collection.immutable.SortedMap
+
 @Configuration
-class AuthManagerConfig @Autowired()(
-  // TODO make it autowired but only if in config AD LDAP provider is set up (#28)
-  usersConfig: UsersConfig,
-  adLDAPConfig: ActiveDirectoryLDAPConfig
-){
+class AuthManagerConfig{
+
+  @Autowired(required = false)
+  private val usersConfig: UsersConfig = null
+  @Autowired (required = false)
+  private val adLDAPConfig: ActiveDirectoryLDAPConfig = null
 
   @Bean
   def authManager(http: HttpSecurity): AuthenticationManager = {
-    val authenticationManagerBuilder = http.getSharedObject(classOf[AuthenticationManagerBuilder])
-
-    // TODO: take which providers and in which order to use from config (#28)
-    authenticationManagerBuilder
-      // if it is not null, on auth failure infinite recursion happens
-      .parentAuthenticationManager(null)
-      // currently, comment out or reorder the auth providers you want to use - #28
-      .authenticationProvider(
-        new ConfigUsersAuthenticationProvider(usersConfig)
-      )
-      .authenticationProvider(
-        new ActiveDirectoryLDAPAuthenticationProvider(adLDAPConfig)
-      )
-      .build
+
+    val authenticationManagerBuilder = http.getSharedObject(classOf[AuthenticationManagerBuilder]).parentAuthenticationManager(null)
+    val configs: Array[DynamicAuthOrder] = Array(usersConfig, adLDAPConfig)
+    val orderedProviders = createProviders(configs)
+
+    if(orderedProviders.isEmpty)
+      throw new Exception("No authentication method enabled in config")
```

Comment:
Just thinking -- here (and maybe below with the IllegalStateException), what about using an appropriate Exception type? Perhaps reusing the ConfigValidationException? Does it fit or do we need to create another type?

Reply:
Used the ConfigValidException. Makes sense to use since it's config related